migrator/replica/postgres/db_replica_manager_impl.go

```diff
@@ -0,0 +1,279 @@
+package postgres
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/migrator/replica/metadata"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/migrations"
+	"github.com/stackrox/rox/pkg/postgres/pgadmin"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/pkg/version"
+)
+
+// DBReplicaManagerImpl - scans and manage database replicas within central.
+type DBReplicaManagerImpl struct {
+	ReplicaMap           map[string]*metadata.DBReplica
+	forceRollbackVersion string
+	adminConfig          *pgxpool.Config
+	sourceMap            map[string]string
+}
+
+// New - returns a new ready-to-use store.
+func New(forceVersion string, adminConfig *pgxpool.Config, sourceMap map[string]string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		ReplicaMap:           make(map[string]*metadata.DBReplica),
+		forceRollbackVersion: forceVersion,
+		adminConfig:          adminConfig,
+		sourceMap:            sourceMap,
+	}
+}
+
+// Scan - checks the persistent data of central and gather the replica information
+// from disk.
+func (d *DBReplicaManagerImpl) Scan() error {
+	replicas := pgadmin.GetDatabaseReplicas(d.adminConfig)
+
+	// We use replicas to collect all db replicas (directory starting with db- or .restore-) matching upgrade or restore pattern.
+	// We maintain replicas with a known link in replicaMap. All unknown replicas are to be removed.
+	replicasToRemove := set.NewStringSet()
+	for _, replica := range replicas {
+		switch name := replica; {
+		case knownReplicas.Contains(name):
+			// Get a short-lived connection for the purposes of checking the version of the replica.
+			pool := pgadmin.GetReplicaPool(d.adminConfig, name)
```

Comment:
It will be desirable if we can switch between databases.

Reply:
I'm not following you.  That pool is to whichever database we are looking at.  central_active, central_backup, central_restore, etc.  Now what I do is move them around such that central itself only ever hits central_active.

Reply:
there is no concept of an alias of a database if that is what you are getting at